pkg/util/pool/pool.go

```diff
@@ -176,3 +176,71 @@ func (b *BatchBytes) Get(sz int) ([]byte, error) {
 	*slab = (*slab)[:len(*slab)+sz]
 	return (*slab)[len(*slab)-sz : len(*slab) : len(*slab)], nil
 }
+
+// SlabPool is NOT concurrency safe.
+type SlabPool[T any] struct {
```

Comment:
Can we write a comment explaining what SlablPool does? You can ask ChatGPT if you want :D

Reply:
This was a very difficult requested. I attempted it in https://github.com/grafana/mimir/pull/3739/commits/a114772a7c73b4bef1a767ce25c30f964741b57b.